docs/source/performance.md

```diff
@@ -0,0 +1,291 @@
+
+Neither of these scenarios have been formally benchmarked to see if performance gains can be made.
+
+### Number of channels a peer is participating in
+
+If a peer has ample CPU and is only participating in a single channel then it's not possible to drive the peer beyond around 65-70% CPU. This is due to the way the peer does serialisation and locking internally. Multiple active channels on a peer will use the extra resource as it increases parallelism in block processing.
+
+In general, ensure that there is a CPU core available for each channel that is running at maximum load. It is not a hard limit however, if the load in each channel is not highly correlated (i.e., not every channel is contenting for resources at the same time), the numbers of channels can exceed the number of CPU cores.
+
+### Total query limit
+
+A peer limits the total number of records a range or JSON (rich) query will return in order to avoid runaway situations. This is configured in the core.yaml file of a peer:
+
+```yaml
+
+ledger:
+  state:
+    # Limit on the number of records to return per query
+    totalQueryLimit: 100000
+```
+
+100000 is the default that Fabric provides in the sample core.yaml and in the test docker images. If you are exceeding this limit, you can increase this value, however you should consider alternative designs that don't require such extensive query scans.
+
+### Concurrency limits
+
+The peer has some limits to ensure a peer cannot be overwhelmed by excessive concurrent client requests:
+
+```yaml
+peer:
+# Limits is used to configure some internal resource limits.
+    limits:
+        # Concurrency limits the number of concurrently running requests to a service on each peer.
+        # Currently this option is only applied to endorser service and deliver service.
+        # When the property is missing or the value is 0, the concurrency limit is disabled for the service.
+        concurrency:
+            # endorserService limits concurrent requests to endorser service that handles chaincode deployment, query and invocation,
+            # including both user chaincodes and system chaincodes.
+            endorserService: 2500
+            # deliverService limits concurrent event listeners registered to deliver service for blocks and transaction events.
+            deliverService: 2500
+            # gatewayService limits concurrent requests to gateway service that handles the submission and evaluation of transactions.
+            gatewayService: 500
+```
+
+The new peer Gateway service introduced a limit in the `gatewayService` with a default of 500. However, this may restrict the TPS of a network so you may need to increase this value to allow more concurrent requests. Good results have been seen with 20000 concurrent requests, higher values may be counter-productive.
```

Comment:
> The new peer Gateway service
it's new now, won't stay new very soon. I'd rather point out in which release it was introduced